src/features/room/sagas/create.saga.ts

```diff
@@ -90,24 +98,27 @@ function* onCreateAction({
         } as Partial<StreamMetadata> & { id: string } & Record<'extensions', Record<'thechat.eth', RoomMetadata>>)
 
         if (privacy === PrivacySetting.TokenGated) {
-            yield put(
-                TokenGatedRoomAction.create({
-                    owner,
-                    tokenAddress: metadata.tokenAddress!,
-                    roomId: stream.id,
-                    tokenIds: metadata.tokenIds
-                        ? metadata.tokenIds.map((id) => BigNumber.from(id).toString())
-                        : [],
-                    minRequiredBalance: metadata.minRequiredBalance?.toString(),
-                    provider,
-                    streamrClient,
-                    tokenType: metadata.tokenType!,
-                    stakingEnabled: metadata.stakingEnabled!,
-                })
-            )
+            yield createTokenGatePolicy({
```

Comment:
@bearni95 `createTokenGatePolicy` uses `spawn` (it's detached). Behaves in a similar way to `put` but doesn't require the `takeEvery` setup.

As you can see below, I got rid of the conditionally present `tokenAddress`, and so on.